sdk/monitor/monitor-opentelemetry/src/shared/config.ts

```diff
@@ -91,7 +91,9 @@ export class InternalConfig implements AzureMonitorOpenTelemetryOptions {
         options.instrumentationOptions,
       );
       this.resource = Object.assign(this.resource, options.resource);
-      this.samplingRatio = options.samplingRatio || this.samplingRatio;
+      if (typeof options.samplingRatio === "number") {
```

Comment:
You should check if it's also a valid number, for example not allowing `NaN`, `Infinite`, etc with something like this:

```js
function isNumber(value) {
  return Number.isFinite(value);
}
```